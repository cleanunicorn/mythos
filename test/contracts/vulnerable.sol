pragma solidity ^0.5.3;

contract Vulnerable {
    uint256 public n = 2^250;

    function f() public {
        selfdestruct(msg.sender);
    }

    function a() public {
        n = n * 2;
    }
}
