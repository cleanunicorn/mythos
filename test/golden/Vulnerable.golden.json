{
  "compiled": {
    "contracts": {
      "test/contracts/vulnerable.sol": {
        "Vulnerable": {
          "abi": [
            {
              "constant": false,
              "inputs": [],
              "name": "a",
              "outputs": [],
              "payable": false,
              "stateMutability": "nonpayable",
              "type": "function"
            },
            {
              "constant": false,
              "inputs": [],
              "name": "f",
              "outputs": [],
              "payable": false,
              "stateMutability": "nonpayable",
              "type": "function"
            },
            {
              "constant": true,
              "inputs": [],
              "name": "n",
              "outputs": [
                {
                  "name": "",
                  "type": "uint256"
                }
              ],
              "payable": false,
              "stateMutability": "view",
              "type": "function"
            }
          ],
          "devdoc": {
            "methods": {}
          },
          "evm": {
            "assembly": "    /* \"test/contracts/vulnerable.sol\":25:197  contract Vulnerable {... */\n  mstore(0x40, 0x80)\n    /* \"test/contracts/vulnerable.sol\":70:75  2^250 */\n  0xf8\n    /* \"test/contracts/vulnerable.sol\":51:75  uint256 public n = 2^250 */\n  0x00\n  sstore\n    /* \"test/contracts/vulnerable.sol\":25:197  contract Vulnerable {... */\n  callvalue\n    /* \"--CODEGEN--\":8:17   */\n  dup1\n    /* \"--CODEGEN--\":5:7   */\n  iszero\n  tag_1\n  jumpi\n    /* \"--CODEGEN--\":30:31   */\n  0x00\n    /* \"--CODEGEN--\":27:28   */\n  dup1\n    /* \"--CODEGEN--\":20:32   */\n  revert\n    /* \"--CODEGEN--\":5:7   */\ntag_1:\n    /* \"test/contracts/vulnerable.sol\":25:197  contract Vulnerable {... */\n  pop\n  dataSize(sub_0)\n  dup1\n  dataOffset(sub_0)\n  0x00\n  codecopy\n  0x00\n  return\nstop\n\nsub_0: assembly {\n        /* \"test/contracts/vulnerable.sol\":25:197  contract Vulnerable {... */\n      mstore(0x40, 0x80)\n      callvalue\n        /* \"--CODEGEN--\":8:17   */\n      dup1\n        /* \"--CODEGEN--\":5:7   */\n      iszero\n      tag_1\n      jumpi\n        /* \"--CODEGEN--\":30:31   */\n      0x00\n        /* \"--CODEGEN--\":27:28   */\n      dup1\n        /* \"--CODEGEN--\":20:32   */\n      revert\n        /* \"--CODEGEN--\":5:7   */\n    tag_1:\n        /* \"test/contracts/vulnerable.sol\":25:197  contract Vulnerable {... */\n      pop\n      jumpi(tag_2, lt(calldatasize, 0x04))\n      div(calldataload(0x00), 0x0100000000000000000000000000000000000000000000000000000000)\n      0x0dbe671f\n      dup2\n      eq\n      tag_3\n      jumpi\n      dup1\n      0x26121ff0\n      eq\n      tag_4\n      jumpi\n      dup1\n      0x2e52d606\n      eq\n      tag_5\n      jumpi\n    tag_2:\n      0x00\n      dup1\n      revert\n        /* \"test/contracts/vulnerable.sol\":149:195  function a() public {... */\n    tag_3:\n      tag_6\n      tag_7\n      jump\t// in\n    tag_6:\n      stop\n        /* \"test/contracts/vulnerable.sol\":82:143  function f() public {... */\n    tag_4:\n      tag_6\n      tag_9\n      jump\t// in\n        /* \"test/contracts/vulnerable.sol\":51:75  uint256 public n = 2^250 */\n    tag_5:\n      tag_10\n      tag_11\n      jump\t// in\n    tag_10:\n      0x40\n      dup1\n      mload\n      swap2\n      dup3\n      mstore\n      mload\n      swap1\n      dup2\n      swap1\n      sub\n      0x20\n      add\n      swap1\n      return\n        /* \"test/contracts/vulnerable.sol\":149:195  function a() public {... */\n    tag_7:\n        /* \"test/contracts/vulnerable.sol\":183:184  n */\n      0x00\n      dup1\n      sload\n        /* \"test/contracts/vulnerable.sol\":187:188  2 */\n      0x02\n        /* \"test/contracts/vulnerable.sol\":183:188  n * 2 */\n      mul\n        /* \"test/contracts/vulnerable.sol\":179:188  n = n * 2 */\n      swap1\n      sstore\n        /* \"test/contracts/vulnerable.sol\":149:195  function a() public {... */\n      jump\t// out\n        /* \"test/contracts/vulnerable.sol\":82:143  function f() public {... */\n    tag_9:\n        /* \"test/contracts/vulnerable.sol\":125:135  msg.sender */\n      caller\n        /* \"test/contracts/vulnerable.sol\":112:136  selfdestruct(msg.sender) */\n      selfdestruct\n        /* \"test/contracts/vulnerable.sol\":51:75  uint256 public n = 2^250 */\n    tag_11:\n      sload(0x00)\n      dup2\n      jump\t// out\n\n    auxdata: 0xa165627a7a723058204988718b74ae76ce71d4c4f2de8e5419a25f1c257569ceb2bf092012fc17d2f60029\n}\n",
            "bytecode": {
              "linkReferences": {},
              "object": "608060405260f860005534801561001557600080fd5b5060c3806100246000396000f3fe6080604052348015600f57600080fd5b50600436106058577c010000000000000000000000000000000000000000000000000000000060003504630dbe671f8114605d57806326121ff01460655780632e52d60614606b575b600080fd5b60636083565b005b6063608e565b60716091565b60408051918252519081900360200190f35b600080546002029055565b33ff5b6000548156fea165627a7a723058204988718b74ae76ce71d4c4f2de8e5419a25f1c257569ceb2bf092012fc17d2f60029",
              "opcodes": "PUSH1 0x80 PUSH1 0x40 MSTORE PUSH1 0xF8 PUSH1 0x0 SSTORE CALLVALUE DUP1 ISZERO PUSH2 0x15 JUMPI PUSH1 0x0 DUP1 REVERT JUMPDEST POP PUSH1 0xC3 DUP1 PUSH2 0x24 PUSH1 0x0 CODECOPY PUSH1 0x0 RETURN INVALID PUSH1 0x80 PUSH1 0x40 MSTORE CALLVALUE DUP1 ISZERO PUSH1 0xF JUMPI PUSH1 0x0 DUP1 REVERT JUMPDEST POP PUSH1 0x4 CALLDATASIZE LT PUSH1 0x58 JUMPI PUSH29 0x100000000000000000000000000000000000000000000000000000000 PUSH1 0x0 CALLDATALOAD DIV PUSH4 0xDBE671F DUP2 EQ PUSH1 0x5D JUMPI DUP1 PUSH4 0x26121FF0 EQ PUSH1 0x65 JUMPI DUP1 PUSH4 0x2E52D606 EQ PUSH1 0x6B JUMPI JUMPDEST PUSH1 0x0 DUP1 REVERT JUMPDEST PUSH1 0x63 PUSH1 0x83 JUMP JUMPDEST STOP JUMPDEST PUSH1 0x63 PUSH1 0x8E JUMP JUMPDEST PUSH1 0x71 PUSH1 0x91 JUMP JUMPDEST PUSH1 0x40 DUP1 MLOAD SWAP2 DUP3 MSTORE MLOAD SWAP1 DUP2 SWAP1 SUB PUSH1 0x20 ADD SWAP1 RETURN JUMPDEST PUSH1 0x0 DUP1 SLOAD PUSH1 0x2 MUL SWAP1 SSTORE JUMP JUMPDEST CALLER SELFDESTRUCT JUMPDEST PUSH1 0x0 SLOAD DUP2 JUMP INVALID LOG1 PUSH6 0x627A7A723058 KECCAK256 0x49 DUP9 PUSH18 0x8B74AE76CE71D4C4F2DE8E5419A25F1C2575 PUSH10 0xCEB2BF092012FC17D2F6 STOP 0x29 ",
              "sourceMap": "25:172:0:-;;;70:5;51:24;;25:172;8:9:-1;5:2;;;30:1;27;20:12;5:2;25:172:0;;;;;;;"
            },
            "deployedBytecode": {
              "linkReferences": {},
              "object": "6080604052348015600f57600080fd5b50600436106058577c010000000000000000000000000000000000000000000000000000000060003504630dbe671f8114605d57806326121ff01460655780632e52d60614606b575b600080fd5b60636083565b005b6063608e565b60716091565b60408051918252519081900360200190f35b600080546002029055565b33ff5b6000548156fea165627a7a723058204988718b74ae76ce71d4c4f2de8e5419a25f1c257569ceb2bf092012fc17d2f60029",
              "opcodes": "PUSH1 0x80 PUSH1 0x40 MSTORE CALLVALUE DUP1 ISZERO PUSH1 0xF JUMPI PUSH1 0x0 DUP1 REVERT JUMPDEST POP PUSH1 0x4 CALLDATASIZE LT PUSH1 0x58 JUMPI PUSH29 0x100000000000000000000000000000000000000000000000000000000 PUSH1 0x0 CALLDATALOAD DIV PUSH4 0xDBE671F DUP2 EQ PUSH1 0x5D JUMPI DUP1 PUSH4 0x26121FF0 EQ PUSH1 0x65 JUMPI DUP1 PUSH4 0x2E52D606 EQ PUSH1 0x6B JUMPI JUMPDEST PUSH1 0x0 DUP1 REVERT JUMPDEST PUSH1 0x63 PUSH1 0x83 JUMP JUMPDEST STOP JUMPDEST PUSH1 0x63 PUSH1 0x8E JUMP JUMPDEST PUSH1 0x71 PUSH1 0x91 JUMP JUMPDEST PUSH1 0x40 DUP1 MLOAD SWAP2 DUP3 MSTORE MLOAD SWAP1 DUP2 SWAP1 SUB PUSH1 0x20 ADD SWAP1 RETURN JUMPDEST PUSH1 0x0 DUP1 SLOAD PUSH1 0x2 MUL SWAP1 SSTORE JUMP JUMPDEST CALLER SELFDESTRUCT JUMPDEST PUSH1 0x0 SLOAD DUP2 JUMP INVALID LOG1 PUSH6 0x627A7A723058 KECCAK256 0x49 DUP9 PUSH18 0x8B74AE76CE71D4C4F2DE8E5419A25F1C2575 PUSH10 0xCEB2BF092012FC17D2F6 STOP 0x29 ",
              "sourceMap": "25:172:0:-;;;;8:9:-1;5:2;;;30:1;27;20:12;5:2;25:172:0;;;;;;;;;;;;;;;;;;;;;;;;;;;;;149:46;;;:::i;:::-;;82:61;;;:::i;51:24::-;;;:::i;:::-;;;;;;;;;;;;;;;;149:46;183:1;;;187;183:5;179:9;;149:46::o;82:61::-;125:10;112:24;51;;;;:::o"
            },
            "gasEstimates": {
              "creation": {
                "codeDepositCost": "39000",
                "executionCost": "20099",
                "totalCost": "59099"
              },
              "external": {
                "a()": "20341",
                "f()": "30139",
                "n()": "422"
              }
            },
            "legacyAssembly": {
              ".code": [
                {
                  "begin": 25,
                  "end": 197,
                  "name": "PUSH",
                  "value": "80"
                },
                {
                  "begin": 25,
                  "end": 197,
                  "name": "PUSH",
                  "value": "40"
                },
                {
                  "begin": 25,
                  "end": 197,
                  "name": "MSTORE"
                },
                {
                  "begin": 70,
                  "end": 75,
                  "name": "PUSH",
                  "value": "F8"
                },
                {
                  "begin": 51,
                  "end": 75,
                  "name": "PUSH",
                  "value": "0"
                },
                {
                  "begin": 51,
                  "end": 75,
                  "name": "SSTORE"
                },
                {
                  "begin": 25,
                  "end": 197,
                  "name": "CALLVALUE"
                },
                {
                  "begin": 8,
                  "end": 17,
                  "name": "DUP1"
                },
                {
                  "begin": 5,
                  "end": 7,
                  "name": "ISZERO"
                },
                {
                  "begin": 5,
                  "end": 7,
                  "name": "PUSH [tag]",
                  "value": "1"
                },
                {
                  "begin": 5,
                  "end": 7,
                  "name": "JUMPI"
                },
                {
                  "begin": 30,
                  "end": 31,
                  "name": "PUSH",
                  "value": "0"
                },
                {
                  "begin": 27,
                  "end": 28,
                  "name": "DUP1"
                },
                {
                  "begin": 20,
                  "end": 32,
                  "name": "REVERT"
                },
                {
                  "begin": 5,
                  "end": 7,
                  "name": "tag",
                  "value": "1"
                },
                {
                  "begin": 5,
                  "end": 7,
                  "name": "JUMPDEST"
                },
                {
                  "begin": 25,
                  "end": 197,
                  "name": "POP"
                },
                {
                  "begin": 25,
                  "end": 197,
                  "name": "PUSH #[$]",
                  "value": "0000000000000000000000000000000000000000000000000000000000000000"
                },
                {
                  "begin": 25,
                  "end": 197,
                  "name": "DUP1"
                },
                {
                  "begin": 25,
                  "end": 197,
                  "name": "PUSH [$]",
                  "value": "0000000000000000000000000000000000000000000000000000000000000000"
                },
                {
                  "begin": 25,
                  "end": 197,
                  "name": "PUSH",
                  "value": "0"
                },
                {
                  "begin": 25,
                  "end": 197,
                  "name": "CODECOPY"
                },
                {
                  "begin": 25,
                  "end": 197,
                  "name": "PUSH",
                  "value": "0"
                },
                {
                  "begin": 25,
                  "end": 197,
                  "name": "RETURN"
                }
              ],
              ".data": {
                "0": {
                  ".auxdata": "a165627a7a723058204988718b74ae76ce71d4c4f2de8e5419a25f1c257569ceb2bf092012fc17d2f60029",
                  ".code": [
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "PUSH",
                      "value": "80"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "PUSH",
                      "value": "40"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "MSTORE"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "CALLVALUE"
                    },
                    {
                      "begin": 8,
                      "end": 17,
                      "name": "DUP1"
                    },
                    {
                      "begin": 5,
                      "end": 7,
                      "name": "ISZERO"
                    },
                    {
                      "begin": 5,
                      "end": 7,
                      "name": "PUSH [tag]",
                      "value": "1"
                    },
                    {
                      "begin": 5,
                      "end": 7,
                      "name": "JUMPI"
                    },
                    {
                      "begin": 30,
                      "end": 31,
                      "name": "PUSH",
                      "value": "0"
                    },
                    {
                      "begin": 27,
                      "end": 28,
                      "name": "DUP1"
                    },
                    {
                      "begin": 20,
                      "end": 32,
                      "name": "REVERT"
                    },
                    {
                      "begin": 5,
                      "end": 7,
                      "name": "tag",
                      "value": "1"
                    },
                    {
                      "begin": 5,
                      "end": 7,
                      "name": "JUMPDEST"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "POP"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "PUSH",
                      "value": "4"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "CALLDATASIZE"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "LT"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "PUSH [tag]",
                      "value": "2"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "JUMPI"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "PUSH",
                      "value": "100000000000000000000000000000000000000000000000000000000"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "PUSH",
                      "value": "0"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "CALLDATALOAD"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "DIV"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "PUSH",
                      "value": "DBE671F"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "DUP2"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "EQ"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "PUSH [tag]",
                      "value": "3"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "JUMPI"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "DUP1"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "PUSH",
                      "value": "26121FF0"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "EQ"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "PUSH [tag]",
                      "value": "4"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "JUMPI"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "DUP1"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "PUSH",
                      "value": "2E52D606"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "EQ"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "PUSH [tag]",
                      "value": "5"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "JUMPI"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "tag",
                      "value": "2"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "JUMPDEST"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "PUSH",
                      "value": "0"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "DUP1"
                    },
                    {
                      "begin": 25,
                      "end": 197,
                      "name": "REVERT"
                    },
                    {
                      "begin": 149,
                      "end": 195,
                      "name": "tag",
                      "value": "3"
                    },
                    {
                      "begin": 149,
                      "end": 195,
                      "name": "JUMPDEST"
                    },
                    {
                      "begin": 149,
                      "end": 195,
                      "name": "PUSH [tag]",
                      "value": "6"
                    },
                    {
                      "begin": 149,
                      "end": 195,
                      "name": "PUSH [tag]",
                      "value": "7"
                    },
                    {
                      "begin": 149,
                      "end": 195,
                      "name": "JUMP",
                      "value": "[in]"
                    },
                    {
                      "begin": 149,
                      "end": 195,
                      "name": "tag",
                      "value": "6"
                    },
                    {
                      "begin": 149,
                      "end": 195,
                      "name": "JUMPDEST"
                    },
                    {
                      "begin": 149,
                      "end": 195,
                      "name": "STOP"
                    },
                    {
                      "begin": 82,
                      "end": 143,
                      "name": "tag",
                      "value": "4"
                    },
                    {
                      "begin": 82,
                      "end": 143,
                      "name": "JUMPDEST"
                    },
                    {
                      "begin": 82,
                      "end": 143,
                      "name": "PUSH [tag]",
                      "value": "6"
                    },
                    {
                      "begin": 82,
                      "end": 143,
                      "name": "PUSH [tag]",
                      "value": "9"
                    },
                    {
                      "begin": 82,
                      "end": 143,
                      "name": "JUMP",
                      "value": "[in]"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "tag",
                      "value": "5"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "JUMPDEST"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "PUSH [tag]",
                      "value": "10"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "PUSH [tag]",
                      "value": "11"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "JUMP",
                      "value": "[in]"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "tag",
                      "value": "10"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "JUMPDEST"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "PUSH",
                      "value": "40"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "DUP1"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "MLOAD"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "SWAP2"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "DUP3"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "MSTORE"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "MLOAD"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "SWAP1"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "DUP2"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "SWAP1"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "SUB"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "PUSH",
                      "value": "20"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "ADD"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "SWAP1"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "RETURN"
                    },
                    {
                      "begin": 149,
                      "end": 195,
                      "name": "tag",
                      "value": "7"
                    },
                    {
                      "begin": 149,
                      "end": 195,
                      "name": "JUMPDEST"
                    },
                    {
                      "begin": 183,
                      "end": 184,
                      "name": "PUSH",
                      "value": "0"
                    },
                    {
                      "begin": 183,
                      "end": 184,
                      "name": "DUP1"
                    },
                    {
                      "begin": 183,
                      "end": 184,
                      "name": "SLOAD"
                    },
                    {
                      "begin": 187,
                      "end": 188,
                      "name": "PUSH",
                      "value": "2"
                    },
                    {
                      "begin": 183,
                      "end": 188,
                      "name": "MUL"
                    },
                    {
                      "begin": 179,
                      "end": 188,
                      "name": "SWAP1"
                    },
                    {
                      "begin": 179,
                      "end": 188,
                      "name": "SSTORE"
                    },
                    {
                      "begin": 149,
                      "end": 195,
                      "name": "JUMP",
                      "value": "[out]"
                    },
                    {
                      "begin": 82,
                      "end": 143,
                      "name": "tag",
                      "value": "9"
                    },
                    {
                      "begin": 82,
                      "end": 143,
                      "name": "JUMPDEST"
                    },
                    {
                      "begin": 125,
                      "end": 135,
                      "name": "CALLER"
                    },
                    {
                      "begin": 112,
                      "end": 136,
                      "name": "SELFDESTRUCT"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "tag",
                      "value": "11"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "JUMPDEST"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "PUSH",
                      "value": "0"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "SLOAD"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "DUP2"
                    },
                    {
                      "begin": 51,
                      "end": 75,
                      "name": "JUMP",
                      "value": "[out]"
                    }
                  ]
                }
              }
            },
            "methodIdentifiers": {
              "a()": "0dbe671f",
              "f()": "26121ff0",
              "n()": "2e52d606"
            }
          },
          "metadata": "{\"compiler\":{\"version\":\"0.5.3+commit.10d17f24\"},\"language\":\"Solidity\",\"output\":{\"abi\":[{\"constant\":false,\"inputs\":[],\"name\":\"a\",\"outputs\":[],\"payable\":false,\"stateMutability\":\"nonpayable\",\"type\":\"function\"},{\"constant\":false,\"inputs\":[],\"name\":\"f\",\"outputs\":[],\"payable\":false,\"stateMutability\":\"nonpayable\",\"type\":\"function\"},{\"constant\":true,\"inputs\":[],\"name\":\"n\",\"outputs\":[{\"name\":\"\",\"type\":\"uint256\"}],\"payable\":false,\"stateMutability\":\"view\",\"type\":\"function\"}],\"devdoc\":{\"methods\":{}},\"userdoc\":{\"methods\":{}}},\"settings\":{\"compilationTarget\":{\"test/contracts/vulnerable.sol\":\"Vulnerable\"},\"evmVersion\":\"byzantium\",\"libraries\":{},\"optimizer\":{\"enabled\":true,\"runs\":200},\"remappings\":[]},\"sources\":{\"test/contracts/vulnerable.sol\":{\"keccak256\":\"0x3036e6836b6b270cf834c24665807c9581bd7fc890a4dc3126646fde7e91874b\",\"urls\":[\"bzzr://65e245004085b4b9cc5cc3a1e241f7106b8ce6d922207d058c026517368bcbdc\"]}},\"version\":1}",
          "userdoc": {
            "methods": {}
          }
        }
      }
    },
    "sources": {
      "test/contracts/vulnerable.sol": {
        "id": 0
      }
    }
  },
  "importedFiles": [
    "test/contracts/empty.sol",
    "test/contracts/import.sol",
    "test/contracts/imported.sol",
    "test/contracts/no-pragma.sol",
    "test/contracts/version-interval.sol",
    "test/contracts/vulnerable.sol"
  ]
}